paddle/fluid/framework/gpu_utils.h

```diff
@@ -82,22 +82,24 @@ struct Index3 : DeviceArray<int, 3, 0> {
 };
 
 // Flat index with real dimension
-EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE int FlatTensorIndex(const Index3& index,
-                                                          const Dim3& dims) {
-  int flat_index = index[0];
+template <typename IDX_T = int>
+EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE IDX_T FlatTensorIndex(const Index3& index,
+                                                            const Dim3& dims) {
+  IDX_T flat_index = index[0];
   for (int i = 1; i < 3; i++) {
     flat_index = flat_index * dims[i] + index[i];
   }
   return flat_index;
 }
 
 // Convert index to tensor index with dimension.
+template <typename IDX_T = int>
 EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Index3
-ConvertTensorIndex(int index, const Dim3& dims) {
+ConvertTensorIndex(IDX_T index, const Dim3& dims) {
   Index3 tensor_index;
   for (int i = 2; i >= 0; i--) {
-    int new_index = index / dims[i];
-    tensor_index[i] = index - dims[i] * new_index;
+    IDX_T new_index = index / dims[i];
+    tensor_index[i] = static_cast<int>(index - dims[i] * new_index);
```

Comment:
Why cast to `int`?

Reply:
The `tensor_index` is an instance of `Index3`, which is int actually. Just avoid some inexplicit casts.

Reply:
Seems like it should be fixed further (not in this PR)... This PR can be merged right now.

Reply:
Just remove this cast? Let int64 be implicit cast to int32 when IDX_T is int64?

Reply:
No. I mean `Index3` might be `Index3<IDX_T>`.

Reply:
Probably in the future when encounter problems 😂, for current usage, use int for `Index3` is enough.

Reply:
Agree so far.